spec/std/uri_spec.cr

```diff
@@ -111,35 +212,21 @@ describe "URI" do
-    it { URI.new("http", "www.example.com", 80).to_s.should eq("http://www.example.com") }
-    it do
-      u = URI.new("http", "www.example.com")
-      u.user = "alice"
-      u.to_s.should eq("http://alice@www.example.com")
-      u.password = "s3cr3t"
-      u.to_s.should eq("http://alice:s3cr3t@www.example.com")
-    end
-    it do
-      u = URI.new("http", "www.example.com")
-      u.user = ":D"
-      u.to_s.should eq("http://%3AD@www.example.com")
-      u.password = "@_@"
-      u.to_s.should eq("http://%3AD:%40_%40@www.example.com")
-    end
+    it { URI.new("http", "www.example.com", 80).to_s.should eq("http://www.example.com:80") }
+    it { URI.new("http", "www.example.com", user: "alice").to_s.should eq("http://alice@www.example.com") }
+    it { URI.new("http", "www.example.com", user: "alice", password: "s3cr3t").to_s.should eq("http://alice:s3cr3t@www.example.com") }
+    it { URI.new("http", "www.example.com", user: ":D").to_s.should eq("http://%3AD@www.example.com") }
+    it { URI.new("http", "www.example.com", user: ":D", password: "@_@").to_s.should eq("http://%3AD:%40_%40@www.example.com") }
     it { URI.new("http", "www.example.com", user: "@al:ce", password: "s/cr3t").to_s.should eq("http://%40al%3Ace:s%2Fcr3t@www.example.com") }
     it { URI.new("http", "www.example.com", fragment: "top").to_s.should eq("http://www.example.com#top") }
-    it { URI.new("http", "www.example.com", 80, "/hello").to_s.should eq("http://www.example.com/hello") }
-    it { URI.new("http", "www.example.com", 80, "/hello", "a=1").to_s.should eq("http://www.example.com/hello?a=1") }
-    it { URI.new("mailto", opaque: "foo@example.com").to_s.should eq("mailto:foo@example.com") }
-
-    it "removes default port" do
-      URI.new("http", "www.example.com", 80).to_s.should eq("http://www.example.com")
-      URI.new("https", "www.example.com", 443).to_s.should eq("https://www.example.com")
-      URI.new("ftp", "www.example.com", 21).to_s.should eq("ftp://www.example.com")
-      URI.new("sftp", "www.example.com", 22).to_s.should eq("sftp://www.example.com")
-      URI.new("ldap", "www.example.com", 389).to_s.should eq("ldap://www.example.com")
-      URI.new("ldaps", "www.example.com", 636).to_s.should eq("ldaps://www.example.com")
-    end
+    it { URI.new("http", "www.example.com", 80, "/hello").to_s.should eq("http://www.example.com:80/hello") }
+    it { URI.new("http", "www.example.com", 80, "/hello", "a=1").to_s.should eq("http://www.example.com:80/hello?a=1") }
+    it { URI.new("mailto", path: "foo@example.com").to_s.should eq("mailto:foo@example.com") }
+    it { URI.new("file", path: "/foo.html").to_s.should eq("file:/foo.html") }
+    it { URI.new("file", path: "foo.html").to_s.should eq("file:foo.html") }
+    it { URI.new("file", host: "host", path: "foo.html").to_s.should eq("file://host/foo.html") }
+    it { URI.new(path: "//foo").to_s.should eq("/.//foo") }
```

Comment:
That's seems odd, why the prefix?

Reply:
An URI starting with double slash is considered relative to the scheme. Thus `//foo` would resolve to a URI with authority `foo` and empty scheme and path.

Reply:
Just now I've noticed it's the path being set and not a the full uri, thanks for explanation.